### ISSUE TITLE
Incorrect "Unused local variable" inspection

### ISSUE THREAD

**Issue by szhorvat** [halirutan/Mathematica-IntelliJ-Plugin#139]
The "Unused local variable" inspection sometimes triggers incorrectly for a `Module` variable that is used as a value (i.e. right-hand-side of `=`) in the variable list of a second enclosed `Module`.

Code to reproduce it (probably not quite minimal, but almost):

```mma
fun[opt : OptionsPattern[]] :=
    Module[{options = Association@Flatten[{opt}]},
      Module[{foo = options["boo"]},
        foo
      ]
    ]
```

Screenshot:

![image](https://user-images.githubusercontent.com/1212871/67375886-8d339680-f583-11e9-8dfc-0bd37ad3b360.png)

----

Update with shorter examples.

The following trigger the inspection incorrectly:

```mma
Module[{b},
  Module[{a = b},
    a; b
  ]
]
```

```mma
Block[{b},
  Block[{a = b},
    a
  ]
]
```

The following *do not* trigger the inspection:

```mma
Function[b,
  Module[{a = b},
    a
  ]
]
```

```mma
Block[{b},
  Module[{a = b},
    a
  ]
]
```

The inner and the outer localizing construct must be the same for the inspection to trigger.

**Comment by halirutan** [comment]
Yep, when shit hits the fan... This will take another session to fix, but I identified the problem.
